torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,161 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    is_alias_to_base: bool = False
```

Comment:
nit: we can do better than these typings, Optional[Tuple[int]] for size/stride, Optional[int] for storage offset

Reply:
done

Reply:
actaully got complains about those form mypy 
ex:
   Argument 3 to "as_strided" has incompatible type "Any | None"; expected
    "Sequence[int | SymInt]"
        Argument 2 to "as_strided" has incompatible type "tuple[int] | None";
    expected "Sequence[int | SymInt]"
    will just keep them Any for now